aiohttp/connector.py

```diff
@@ -470,25 +479,33 @@ def _get(self, key):
         del self._conns[key]
         return None
 
+    def _release_key_waiter(self, key, waiters):
+        if not waiters:
+            return False
+
+        waiter = waiters.pop(0)
+        waiter.set_result(None)
+
+        if not waiters:
+            del self._waiters[key]
+
+        return True
+
     def _release_waiter(self):
         # always release only one waiter
 
         if self._limit:
             # if we have limit and we have available
             if self._limit - len(self._acquired) > 0:
                 for key, waiters in self._waiters.items():
-                    if waiters:
-                        if not waiters[0].done():
```

Comment:
with the new model we can guarantee that there are only active waiters in the list so we can greatly simplify this to always popping and notifying the first item in the list